c10/util/Logging.cpp

```diff
@@ -356,3 +390,56 @@ MessageLogger::~MessageLogger() {
 } // namespace c10
 
 #endif // !C10_USE_GLOG
+
+namespace c10 {
+namespace detail {
+namespace {
+
+void unsafeSetLogLevelFlagFromEnv() {
+  const char* level_str = std::getenv("TORCH_CPP_MIN_LOG_LEVEL");
+
+  // Not set, fallback to the default level (i.e. WARNING).
+  if (level_str == nullptr) {
+    return;
+  }
+
+  std::string level{level_str};
+  if (level.empty()) {
+    return;
+  }
+
+  if (level == "0" || level == "INFO" || level == "info") {
+    FLAGS_caffe2_log_level = 0;
+
+    return;
+  }
+  if (level == "1" || level == "WARNING" || level == "warning") {
+    FLAGS_caffe2_log_level = 1;
+
+    return;
+  }
+  if (level == "2" || level == "ERROR" || level == "error") {
+    FLAGS_caffe2_log_level = 2;
+
+    return;
+  }
+  if (level == "3" || level == "FATAL" || level == "fatal") {
+    FLAGS_caffe2_log_level = 3;
+
+    return;
+  }
+
+  std::cerr << "`TORCH_CPP_MIN_LOG_LEVEL` environment variable cannot be parsed. Valid values are "
+               "`INFO`, `WARNING`, `ERROR`, and `FATAL` or their numerical equivalents `0`, `1`, "
+               "`2`, and `3`." << std::endl;
+}
+
+void setLogLevelFlagFromEnv() {
+  static std::once_flag f{};
+
+  std::call_once(f, unsafeSetLogLevelFlagFromEnv);
+}
```

Comment:
How about something like the follow (or even simple `int FLAGS_caffe2_log_level = getLogLevelFromEnv();` when this flag is defined) 
```suggestion
namespace {
  struct LogLevelInitializer {
    LogLevelInitializer() {
        unsafeSetLogLevelFlagFromEnv();
    }
  } _log_level_initializer;
}
```

Reply:
Please see the second point in my comment above.